### PR TITLE
various fixes for 0.6

### PR DESCRIPTION
**Pull request by tlnagy**
Needed for https://github.com/GiovineItalia/Gadfly.jl/issues/930

### CODE REVIEW

File: src/boundingbox.jl [hunk 1]
```diff
@@ -26,11 +26,11 @@ BoundingBox(width, height, depth) = BoundingBox(0mm, 0mm, 0mm, width, height, de
 
 isabsolute{P1, P2}(b::BoundingBox{P1, P2}) = isabsolute(b.x0) && isabsolute(b.a)
 
-typealias AbsoluteBox{N}   BoundingBox{NTuple{N, Length{:mm, Float64}},
+const AbsoluteBox{N} = BoundingBox{NTuple{N, Length{:mm, Float64}},
```
Comment (shashi) [inline src/boundingbox.jl]: This needs a `@compat` to work on 0.5 right?

File: src/point.jl [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
 
-
-#typealias Point Tuple{Vararg{Measure}}
-#typealias Point2D Tuple{Measure, Measure}
-#typealias Point3D Tuple{Measure, Measure, Measure}
-
-typealias Vec{N} NTuple{N, Measure}
-typealias Vec2 Vec{2}
-typealias Vec3 Vec{3}
+const Vec{N} = NTuple{N, Measure}
```
Comment (shashi) [inline src/point.jl]: Same here

File: src/point.jl [hunk 3]
```diff
@@ -17,9 +12,9 @@ isabsolute{N}(p::NTuple{N, AbsoluteLength}) = true
 #Vec(x::Measure, y::Measure) = Vec{2, Measure}((x, y))
 #Vec() = Vec(0mm, 0mm)
 
-typealias AbsoluteVec{N} NTuple{N, Length{:mm, Float64}}
-typealias AbsoluteVec2 AbsoluteVec{2}
-typealias AbsoluteVec3 AbsoluteVec{3}
+const AbsoluteVec{N} = NTuple{N, Length{:mm, Float64}}
```
Comment (shashi) [inline src/point.jl]: And here

File: src/boundingbox.jl (hunk 1 above)
Comment (tlnagy) [reply src/boundingbox.jl]: Yeah, working on it rn

File: src/point.jl (hunk 2 above)
Comment (bjarthur) [inline src/point.jl]: i think an `@compat` is needed when both sides are parameterized

**Comment by tlnagy** [comment]
Thank you all for the quick feedback, it looks like everything is passing on both 0.5 and 0.6. I'll wait for the mac os builds to pass before merging.

File: src/boundingbox.jl [hunk 4]
```diff
@@ -26,11 +26,11 @@ BoundingBox(width, height, depth) = BoundingBox(0mm, 0mm, 0mm, width, height, de
 
 isabsolute{P1, P2}(b::BoundingBox{P1, P2}) = isabsolute(b.x0) && isabsolute(b.a)
 
-typealias AbsoluteBox{N}   BoundingBox{NTuple{N, Length{:mm, Float64}},
+@compat const AbsoluteBox{N} = BoundingBox{NTuple{N, Length{:mm, Float64}},
                                        NTuple{N, Length{:mm, Float64}}}
-typealias Absolute2DBox    BoundingBox{Tuple{AbsoluteLength, AbsoluteLength},
+@compat const Absolute2DBox =  BoundingBox{Tuple{AbsoluteLength, AbsoluteLength},
```
Comment (bjarthur) [inline src/boundingbox.jl]: might not be important, but i don't think `@compat` is needed here since the left hand side is not parameterized

Comment (tlnagy) [reply src/boundingbox.jl]: Let me try without it, I guess we should limit the number of calls to `@compat` to be safe

Comment (bjarthur) [reply src/boundingbox.jl]: if it works without, then there are a few other places it can be removed as well

Comment (tlnagy) [reply src/boundingbox.jl]: Yeah, it seems to be working. I'll remove it for all nonparamterized typealiasing calls

**Comment by tlnagy** [comment]
All green, LGTM

**Pull request merged**